cluster-autoscaler/go.mod

```diff
@@ -1,6 +1,6 @@
 module k8s.io/autoscaler/cluster-autoscaler
 
-go 1.21.3
+go 1.21
```

Comment:
Pulling in same revert of change here as in https://github.com/kubernetes/autoscaler/pull/6317